src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -402,7 +402,7 @@ public abstract class Instrument<T> : Instrument where T : struct
         public ReadOnlySpan<System.Collections.Generic.KeyValuePair<string, object?>> Tags { get { throw null; } }
         public T Value { get { throw null; } }
     }
-    public delegate void MeasurementCallback<T>(Instrument instrument, T measurement, ReadOnlySpan<System.Collections.Generic.KeyValuePair<string, object?>> tags, object? state);
+    public delegate void MeasurementCallback<T>(Instrument instrument, T measurement, ReadOnlySpan<System.Collections.Generic.KeyValuePair<string, object?>> tags, object? state) where T : struct;
```

Comment:
Are all of these internal/non-public types?

Adding a constraint is a binary breaking, so just wanting to double check. -- notably, changing `struct` to `unmanaged` is binary compatible, but may not be source compatible

Reply:
It looks to me like they're all impacting the ref assembly, so users calling these methods from unconstrained or less constrained APIs will break on roll forward.

Reply:
These are all public types.   I've marked this change breaking.  The constraints were already present in the runtime so I believe folks would have already observed runtime errors if they were violating the constraint.  
```
System.TypeLoadException: GenericArguments[0], 'System.String', on 'System.Diagnostics.Metrics.MeasurementCallback`1[T]' violates the constraint of type parameter 'T'.
   at Program.<Main>$(String[] args)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
```

If we don't want to add the source breaking for any of these we should remove the runtime constraint.

Reply:
> The constraints were already present in the runtime so I believe folks would have already observed runtime errors if they were violating the constraint.

👍. I was definitely more concerned about binary breaks, I think source breaks are fine particularly given that a TypeLoadException was already occurring.